src/main/java/chess/controller/ChessGameController.java

```diff
@@ -0,0 +1,73 @@
+package chess.controller;
+
+import chess.domain.chessboard.ChessBoard;
+import chess.domain.chessboard.Square;
+import chess.domain.chessgame.ChessGame;
+import chess.domain.chessgame.GameCommand;
+import chess.dto.ChessBoardDto;
+import chess.view.InputView;
+import chess.view.OutputView;
+import java.util.List;
+import java.util.function.Supplier;
+
+public class ChessGameController {
+
+    private static final Integer GAME_COMMAND_INDEX = 0;
+
+    public void run() {
+        OutputView.printStartMessage();
+        gameStart();
+    }
+
+    private void gameStart() {
+        List<String> input = repeatUntilSuccess(InputView::requestGameCommand);
+        String inputCommand = input.get(GAME_COMMAND_INDEX);
+        GameCommand gameCommand = GameCommand.findGameCommand(inputCommand);
+        try {
+            executeGameUntilEnd(gameCommand, input);
+        } catch (IllegalArgumentException e) {
+            OutputView.printErrorMessage(e.getMessage());
+            gameStart();
+        }
+    }
+
+    private void executeGameUntilEnd(GameCommand initialGameCommand, List<String> initialInput) {
+        ChessBoard chessBoard = new ChessBoard();
+        GameCommand gameCommand = initialGameCommand;
+        List<String> input = initialInput;
+        while (GameCommand.isNotFinishedGame(gameCommand)) {
```

Comment:
아래와 같은 형태로 메세지를 던지면 좋을것 같습니다.

```suggestion
        while (gameCommand.isNotFinishedGame()) {
```